### DISCUSSION TITLE
t.timeout is ignored in test.before

### DISCUSSION THREAD

**Discussion by fungiboletus** [avajs/ava#2888] [Q&A] [answered]
```typescript
import test from 'ava';
import { setTimeout } from 'timers';

test.before(async (t) => {
  t.timeout(60_000);
  await new Promise<void>((resolve) => setTimeout(() => resolve(), 50_000));
});

test('Works', (t) => {
  t.is(1, 1);
});
```

Running the following code will fail instead of waiting 50 seconds.

t.timeout should perhaps be removed from the TypeScript definition of the t parameter in the test.before callback. But it would also be nice to be able to specify a timeout in a before block.

**Comment by novemberborn** [comment, marked as answer]
Could you elaborate on how the code fails?

In AVA 3, the global timeout (which has a default shorter than 60 seconds) is not extended when you use `t.timeout()`. That's fixed in AVA 4 (pre-release).

**Reply by fungiboletus** [reply]
It's failing because of the shorter timeout, letting me think that the 60 seconds timeout is ignored. Good to know it's fixed in the next major release.
